src/file_view_pool.cpp

```diff
@@ -221,7 +221,7 @@ namespace libtorrent { namespace aux {
 		auto const end = key_view.upper_bound(file_id{st, std::numeric_limits<file_index_t>::max()});
 
 		for (auto it = begin; it != end; ++it)
-			defer_destruction.emplace_back(std::move(it->mapping));
```

Comment:
I would prefer to figure out how to get to a mutable mapping, in order to actually make these moves